src/DotLiquid/StandardFilters.cs

```diff
@@ -166,12 +166,14 @@ public static string H(string input)
         /// <param name="length"></param>
         /// <param name="truncateString"></param>
         /// <returns></returns>
-        public static string Truncate(string input, int length = 50, string truncateString = "...")
+        public static string Truncate(string input, long length = 50, string truncateString = "...")
         {
             if (string.IsNullOrEmpty(input))
                 return input;
 
-            int l = length - truncateString.Length;
+            int lengthAsInt = length < int.MinValue ? int.MinValue : (length > int.MaxValue ? int.MaxValue : (int)length);
+
+            int l = lengthAsInt - truncateString.Length;

```

Comment:
for code readability, this is the right opportunity to rename the variable to some descriptive word or compound words. 'l' and other single letter variable names are poor choices giving little to no information to engineers trying to maintain the code. As the change requires to understand the existing purpose, do rename to something meaningful.

Reply:
Can do. I usually prefer a more descriptive variable names, but didn't think it my place to change existing naming here.